src/julia_internal.h

```diff
@@ -216,6 +219,26 @@ static inline uint64_t cycleclock(void) JL_NOTSAFEPOINT
     int64_t virtual_timer_value;
     __asm__ volatile("mrs %0, cntvct_el0" : "=r"(virtual_timer_value));
     return virtual_timer_value;
+#elif defined(_CPU_ARM_)
+  // V6 is the earliest arch that has a standard cyclecount
+#if (__ARM_ARCH >= 6)
+  uint32_t pmccntr;
+  uint32_t pmuseren;
+  uint32_t pmcntenset;
+  // Read the user mode perf monitor counter access permissions.
+  asm volatile("mrc p15, 0, %0, c9, c14, 0" : "=r"(pmuseren));
+  if (pmuseren & 1) {  // Allows reading perfmon counters for user mode code.
+    asm volatile("mrc p15, 0, %0, c9, c12, 1" : "=r"(pmcntenset));
+    if (pmcntenset & 0x80000000ul) {  // Is it counting?
+      asm volatile("mrc p15, 0, %0, c9, c13, 0" : "=r"(pmccntr));
+      // The counter is set up to count every 64th cycle
+      return (int64_t)(pmccntr) * 64;  // Should optimize to << 6
+    }
+  }
+#endif
+  struct timeval tv;
+  gettimeofday(&tv, NULL);
+  return (int64_t)(tv.tv_sec) * 1000000 + tv.tv_usec;
```

Comment:
Indentation here should be 4 spaces
```suggestion
    // V6 is the earliest arch that has a standard cyclecount
#if (__ARM_ARCH >= 6)
    uint32_t pmccntr;
    uint32_t pmuseren;
    uint32_t pmcntenset;
    // Read the user mode perf monitor counter access permissions.
    asm volatile("mrc p15, 0, %0, c9, c14, 0" : "=r"(pmuseren));
    if (pmuseren & 1) {  // Allows reading perfmon counters for user mode code.
        asm volatile("mrc p15, 0, %0, c9, c12, 1" : "=r"(pmcntenset));
        if (pmcntenset & 0x80000000ul) {  // Is it counting?
            asm volatile("mrc p15, 0, %0, c9, c13, 0" : "=r"(pmccntr));
            // The counter is set up to count every 64th cycle
            return (int64_t)(pmccntr) * 64;  // Should optimize to << 6
        }
    }
#endif
    struct timeval tv;
    gettimeofday(&tv, NULL);
    return (int64_t)(tv.tv_sec) * 1000000 + tv.tv_usec;
```
Also, is `__ARM_ARCH` guaranteed to exist on ARM with both gcc and clang? Would a `defined(__ARM_ARCH)` be worthwhile to add?

Reply:
Should probably also reference the source of the code in a comment

Reply:
Indentation is fixed.

As far as I can tell, all the `cycleclock` is taken from the same location, and there already is a comment indicating this (pointing to https://github.com/google/benchmark/blob/v1.5.0/src/cycleclock.h). Well, that comment is only in the fallback `#else` case; perhaps it should be moved to the top of `cycleclock` resp. into a comment before it? 

As to `__ARM_ARCH`, this is used unguarded in multiple places already. In any case, if it is not defined, then `#if (__ARM_ARCH >= 6)` will evaluate to `#if 0`, which arguably is the desired behaviour?

Reply:
> Also, is `__ARM_ARCH` guaranteed to exist on ARM with both gcc and clang?

It's defined in the [Arm C Language Extensions](https://developer.arm.com/documentation/101028/latest) specification, so one would hope compliant compilers would follow it.

Reply:
Sorry, I indeed missed the final three lines (and based on other surrounding code, the `#if` shouldn't be indented?). Hopefully OK now